### PR TITLE
feat: [CDS-86991]: Yaml simplification for Approval Stage

### PR DESCRIPTION
**Pull request by sarthakkasat**
Screenshot 2023-12-13 at 11 49 00 AM

### CODE REVIEW

File: v1/pipeline/stages/approval/approval-stage-config.yaml [hunk 1]
```diff
@@ -1,10 +1,12 @@
 title: ApprovalStageConfig
 type: object
 required:
-- execution
+- steps
```
Comment (shalini-agr) [inline v1/pipeline/stages/approval/approval-stage-config.yaml:4]: add field rollback_steps as well

File: v1/pipeline.json [hunk 2]
```diff
@@ -3064,81 +3064,56 @@
...
-                "additionalProperties" : {
-                  "type" : "string"
-                }
+              "strategy" : {
+                "description" : "The stage execution strategy.",
+                "$ref" : "#/definitions/pipeline/common/Strategy"
               },
               "type" : {
                 "type" : "string",
                 "enum" : [ "Approval" ]
```
Comment (shalini-agr) [inline v1/pipeline.json:3093]: this should be `approval`

File: v1/pipeline/stages/approval/approval-stage-config.yaml (hunk 1 above)
Comment (shalini-agr) [reply v1/pipeline/stages/approval/approval-stage-config.yaml:4]: not needed in approval

File: v1/pipeline.json (hunk 2 above)
Comment (sarthakkasat) [reply v1/pipeline.json:3093]: Done

File: v1/pipeline/stages/approval/approval-stage-node.yaml [hunk 3]
```diff
@@ -3,65 +3,51 @@ type: object
 required:
 - type
```
Comment (shalini-agr) [inline v1/pipeline/stages/approval/approval-stage-node.yaml:4]: add spec here

File: v1/pipeline/stages/approval/approval-stage-node.yaml [hunk 4]
```diff
@@ -3,65 +3,51 @@ type: object
...
-      pattern: ^<\+input>$
-      minLength: 1
-  tags:
-    type: object
-    additionalProperties:
-      type: string
+    $ref: ../../common/strategy.yaml
+    description: The stage execution strategy.
   type:
     type: string
```
Comment (shalini-agr) [inline v1/pipeline/stages/approval/approval-stage-node.yaml:25]: add description to this field

File: v1/pipeline/stages/approval/approval-stage-node.yaml (hunk 3 above)
Comment (sarthakkasat) [reply v1/pipeline/stages/approval/approval-stage-node.yaml:4]: Done

File: v1/pipeline/stages/approval/approval-stage-node.yaml (hunk 4 above)
Comment (sarthakkasat) [reply v1/pipeline/stages/approval/approval-stage-node.yaml:27]: Done

File: v1/pipeline/stages/approval/approval-stage-node.yaml [hunk 5]
```diff
@@ -2,66 +2,54 @@ title: ApprovalStageNode
 type: object
 required:
 - type
```
Comment (shalini-agr) [inline v1/pipeline/stages/approval/approval-stage-node.yaml:4]: indentation is not correct

File: v1/pipeline/stages/approval/approval-stage-node.yaml [hunk 6]
```diff
@@ -3,65 +3,51 @@ type: object
...
+    type: object
+    description: Variables defines the pipeline variable parameters.
+    additionalProperties:
       oneOf:
-      - $ref: ../../common/number-ng-variable.yaml
-      - $ref: ../../common/secret-ng-variable.yaml
-      - $ref: ../../common/string-ng-variable.yaml
+        - "$ref": ../../common/string-variable.yaml
+        - "$ref": ../../common/number-variable.yaml
+        - "$ref": ../../common/array-variable.yaml
```
[the remark this answers is not in the document]
Comment (sarthakkasat) [reply v1/pipeline/stages/approval/approval-stage-node.yaml:37]: Done

File: v1/pipeline/stages/approval/approval-stage-node.yaml (hunk 5 above)
Comment (sarthakkasat) [reply v1/pipeline/stages/approval/approval-stage-node.yaml]: Done

File: v1/pipeline/stages/approval/approval-stage-node.yaml [hunk 7]
```diff
@@ -3,65 +3,51 @@ type: object
...
-    - type: string
-      pattern: ^<\+input>$
-      minLength: 1
+    description: The stage conditional logic.
+    type: string
+  labels:
+    type: object
+    additionalProperties:
+      type: string
+
```
[the remark this answers is not in the document]
Comment (sarthakkasat) [reply v1/pipeline/stages/approval/approval-stage-node.yaml:51]: Done

**Review by shalini-agr — APPROVED** [review]

**Review by utkarshchoubey-harness — APPROVED** [review]
lgtm!

**Review by archit-harness — APPROVED** [review]

**Pull request merged**